driver-sync/src/main/com/mongodb/client/internal/ChangeStreamIterableImpl.java

```diff
@@ -98,6 +99,12 @@
         return this;
     }
 
+    @Override
+    public ChangeStreamIterable<TResult> showMigrationEvents(final boolean showMigrationEvents) {
+        this.showMigrationEvents = notNull("showMigrationEvents", showMigrationEvents);
```

Comment:
Remove this line.  Doesn't make sense for a primitive value.  What's happening here is that the primitive boolean is being auto-boxed into a `Boolean`.